### ISSUE TITLE
`indent` should check tabs in front of block comments

### ISSUE THREAD

**Issue by ibbignerd** [eslint/eslint#11576]
**What rule do you want to change?**
[indent](https://eslint.org/docs/rules/indent)
**Does this change cause the rule to produce more or fewer warnings?**
More warnings
**How will the change be implemented? (New option, new default behavior, etc.)?**
This could be a new option or modify default behavior. The former would be a non-breaking change and likely preferred.
**Please provide some example code that this change will affect:**

```js
export class Human {
	constructor(name) {
		this.name = name;
	}

	/**
	 * Set the `name` of the class
	 * @param {string} value - value to be set
	 */
	set name(value) {
		this._name = value;
	}

    /**
     * Get the `name` of the class
     * @returns {string}
     */
    get name() {
        return this._name;
    }
}
```

**What does the rule currently do for this code?**
[demo link](https://eslint.org/demo/#eyJ0ZXh0IjoiZXhwb3J0IGNsYXNzIEh1bWFuIHtcclxuICAgIGNvbnN0cnVjdG9yKG5hbWUpIHtcclxuICAgICAgICB0aGlzLm5hbWUgPSBuYW1lO1xyXG4gICAgfVxyXG5cclxuICAgIC8qKlxyXG5cdCAqIFNldCB0aGUgYG5hbWVgIG9mIHRoZSBjbGFzc1xyXG5cdCAqIEBwYXJhbSB7c3RyaW5nfSB2YWx1ZSAtIHZhbHVlIHRvIGJlIHNldFxyXG5cdCAqL1xyXG4gICAgc2V0IG5hbWUodmFsdWUpIHtcclxuICAgICAgICB0aGlzLl9uYW1lID0gdmFsdWU7XHJcbiAgICB9XHJcblxyXG4gICAgLyoqXHJcbiAgICAgKiBHZXQgdGhlIGBuYW1lYCBvZiB0aGUgY2xhc3NcclxuICAgICAqIEByZXR1cm5zIHtzdHJpbmd9XHJcbiAgICAgKi9cclxuICAgIGdldCBuYW1lKCkge1xyXG4gICAgICAgIHJldHVybiB0aGlzLl9uYW1lO1xyXG4gICAgfVxyXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImluZGVudCI6Mn0sImVudiI6e319fQ==). Notice how on lines 7-9 are prepended with tabs, but an error is not thrown. The inverse is true when `["error", "tab"]` is used. 
**What will the rule do after it's changed?**
The rule should check blocked comments and ensure they follow the indent rule properly. 
**Are you willing to submit a pull request to implement this change?**
No.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal. Have you tried the `no-mixed-spaces-and-tabs` rule? I think it generally handles this case instead.

**Comment by ibbignerd** [comment]
I have. We currently have that enabled. The above should never throw for `no-mixed-spaces-and-tabs`.

The root of the issue is that blocked comments aren't checked if they are tabs or spaces. If I want my project to use tabs (`["error", "tab"]`), it should force me to use tabs in front of my blocked comments too. Currently, it does not error when I use spaces instead. 

Notice how the above [demo link](https://eslint.org/demo/#eyJ0ZXh0IjoiZXhwb3J0IGNsYXNzIEh1bWFuIHtcclxuICAgIGNvbnN0cnVjdG9yKG5hbWUpIHtcclxuICAgICAgICB0aGlzLm5hbWUgPSBuYW1lO1xyXG4gICAgfVxyXG5cclxuICAgIC8qKlxyXG5cdCAqIFNldCB0aGUgYG5hbWVgIG9mIHRoZSBjbGFzc1xyXG5cdCAqIEBwYXJhbSB7c3RyaW5nfSB2YWx1ZSAtIHZhbHVlIHRvIGJlIHNldFxyXG5cdCAqL1xyXG4gICAgc2V0IG5hbWUodmFsdWUpIHtcclxuICAgICAgICB0aGlzLl9uYW1lID0gdmFsdWU7XHJcbiAgICB9XHJcblxyXG4gICAgLyoqXHJcbiAgICAgKiBHZXQgdGhlIGBuYW1lYCBvZiB0aGUgY2xhc3NcclxuICAgICAqIEByZXR1cm5zIHtzdHJpbmd9XHJcbiAgICAgKi9cclxuICAgIGdldCBuYW1lKCkge1xyXG4gICAgICAgIHJldHVybiB0aGlzLl9uYW1lO1xyXG4gICAgfVxyXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImluZGVudCI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyfSwiZW52Ijp7fX19) requires spaces, but there are tabs prepending the block comment on lines 7-9

**Comment by platinumazure** [comment]
Hi @ibbignerd, just wanted to leave a note about the demo...

You can pass options in the demo by having your source code use an [inline configuration comment](https://eslint.org/docs/user-guide/configuring#using-configuration-comments). Here's an [example](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW2Vycm9yLCB0YWJdICovXHJcblxyXG5pZiAoZm9vKSB7XHJcbiAgICBiYXIoKTtcclxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==).

Hope this helps! (Sorry, I can't help with the actual issue at the moment.)

**Comment by ibbignerd** [comment]
@not-an-aardvark Did you see my demo that highlights the issue?
